packages/next/src/server/dev/hot-reloader-webpack.ts

```diff
@@ -751,6 +753,22 @@ export default class HotReloaderWebpack implements NextJsHotReloaderInterface {
       isTestMode || this.telemetry.isEnabled
     )
 
+    const nodeDebugType = getNodeDebugType()
+    if (nodeDebugType && !this.devtoolsFrontendUrl) {
+      const debugPort = process.debugPort
+      let debugInfo
+      try {
+        // It requires to use 127.0.0.1 instead of localhost for server-side fetching.
+        const debugInfoList = await fetch(
+          `http://127.0.0.1:${debugPort}/json/list`
+        ).then((res) => res.json())
+        debugInfo = debugInfoList[0]
```

Comment:
a bit of an edge case, but how do we determine which index to access here? If there was a separate inspector running in addition to the Next.js one, I guess we'd potentially open up the wrong one?

Reply:
There will only be 1 available item for current next-server process since it's already using the current debug port, I added a comment for it above in the new commit

Reply:
That's true, but you could have another Next.js process running on a debugger port, in which case how does it know which one to use?

Reply:
It's not able to start the new inspector on new process, it always started with 9229 port and will fail if we try to debug new nextjs app

```
Starting inspector on 127.0.0.1:9229 failed: address already in use
Starting inspector on 127.0.0.1:9230 failed: address already in use
```